snuba/utils/streams/backends/abstract.py

```diff
@@ -1,19 +1,11 @@
+from __future__ import annotations
```

Comment:
Do you have opinions with regards to having all this logic in the __init__ file of the backends module vs having a specific file for the abstractions (this one)?
I generally go for the second when I do not want to import all the abstractions transitively when loading any file in the module (which happens if I put the abstractions in __init__), though it does into seem an issue here. Otherwise I tend to put the abstractions in init so that the import statement are shorter.

Reply:
I don't have a very strong opinion. I guess I slightly prefer keeping the definitions in a separate file (mostly because I never think to look for actual code in `__init__` out of habit) but think it makes sense to import them in `__init__` for reexport by convention. I think I had tried this somewhere else in the past (maybe metrics?) but ran into some mypy warnings and never investigated further. It seems like an easy thing to work around, though: https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-no-implicit-reexport

Reply:
What do you think about exposing these from `snuba.utils.streams`:


```python
from snuba.utils.streams.backends.abstract import Consumer, ConsumerError, Producer
from snuba.utils.streams.types import Message, Partition, Topic

__all__ = [
    "Consumer",
    "ConsumerError",
    "Message",
    "Partition",
    "Producer",
    "Topic",
]
```

Those are pretty much all of the commonly used types that are not directly associated with either a specific backend or processing strategy (which I think should be kept separate — at least as long as there are multiple implementation options.)

I left out `TPayload` from types as that generally shouldn't be imported outside of the streams package itself.

Reply:
Addressing this in a follow up instead.

Reply:
`snuba.utils.streams` seems a good idea to me, even better than exporting them from `backends`
And about the mypy issue, I am not sure, never tried.